cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -191,7 +191,7 @@ public void getAllApps(HttpRequest request, HttpResponder responder,
   public void getAppInfo(HttpRequest request, HttpResponder responder,
                          @PathParam("namespace-id") String namespaceId,
                          @PathParam("app-id") final String appId)
-    throws NamespaceNotFoundException, BadRequestException, ApplicationNotFoundException {
```

Comment:
Since this handler calls `getAppDetail`, which involves authorization. The `authorizationEnforcer` will throw `Exception`.